extra_data/components.py

```diff
@@ -304,6 +329,25 @@ def _select_pulse_indices(pulses, firsts, counts):
         return np.concatenate(positions)
 
     def _make_image_index(self, tids, inner_ids, inner_name='pulse'):
+        """
+        Prepare indices for data per inner coordinate.
```

Comment:
Thanks for adding these! We don't normally bother writing the full lists of parameters etc. for private methods (names beginning with `_`), but there's also no need to remove them now you've added them.

Reply:
Thank you!) Yes, I understand that full docstrings are not required for protected methods, but I added them mostly for myself while trying to understand how the code works.